src/safeds/data/image/containers/_multi_size_image_list.py

```diff
@@ -0,0 +1,542 @@
+                for image_size, image_list in self._image_list_dict.items():
+                    image_list.to_png_files(
+                        [p for i, p in enumerate(path) if self._indices_to_image_size_dict[i] == image_size],
+                    )
+            elif len(path) == self.number_of_sizes:
+                image_list_path: str | Path
+                for image_list_path, image_list in zip(path, self._image_list_dict.values(), strict=False):
+                    image_list.to_png_files(image_list_path)
+            else:
+                raise ValueError(
+                    "The path specified is invalid. Please provide either the path to a directory, a list of paths with one path for each image, or a list of paths with one path per image size.",
+                )
+        else:
+            for image_list in self._image_list_dict.values():
+                image_list.to_png_files(path)
+
+    def to_images(self, indices: list[int] | None = None) -> list[Image]:
+        if indices is None:
+            indices = sorted(self._indices_to_image_size_dict)
+        else:
+            wrong_indices = []
+            for index in indices:
+                if index not in self._indices_to_image_size_dict:
+                    wrong_indices.append(index)
+            if len(wrong_indices) != 0:
+                raise IndexOutOfBoundsError(wrong_indices)
+        images = []
+        for index in indices:
+            images.append(self._image_list_dict[self._indices_to_image_size_dict[index]].get_image(index))
+        return images
+
+    def change_channel(self, channel: int) -> ImageList:
+        image_list = self._clone_without_image_dict()
+        for image_list_key, image_list_original in self._image_list_dict.items():
+            image_list._image_list_dict[image_list_key] = image_list_original.change_channel(channel)
+        return image_list
+
+    def _add_image_tensor(self, image_tensor: Tensor, index: int) -> ImageList:
+        from safeds.data.image.containers._single_size_image_list import _SingleSizeImageList
+
+        if index in self._indices_to_image_size_dict:
+            raise DuplicateIndexError(index)
+
+        image_list = self.clone()._as_multi_size_image_list()
+        size = (image_tensor.size(dim=2), image_tensor.size(dim=1))
+        image_list._indices_to_image_size_dict[index] = size
+
+        if size in self._image_list_dict:
+            image_list._image_list_dict[size] = image_list._image_list_dict[size]._add_image_tensor(image_tensor, index)
+        else:
+            image_list._image_list_dict[size] = _SingleSizeImageList._create_image_list([image_tensor], [index])
+
+        if image_tensor.size(dim=0) != self.channel:
+            image_list = image_list.change_channel(
+                max(image_tensor.size(dim=0), self.channel),
+            )._as_multi_size_image_list()
+
+        return image_list
+
+    def add_images(self, images: list[Image] | ImageList) -> ImageList:
```

Comment:
shouldn't the return type be multi_image_list?
for the most functions in the class